core/transcoder.go

```diff
@@ -27,9 +27,20 @@ type LocalTranscoder struct {
 	workDir string
 }
 
+type UnrecoverableError struct {
```

Comment:
Since all this struct does is wrap an `error` could we simplify by defining `UnrecoverableError` as an `error`?

```golang
type UnrecoverableError error
```

And then if we need to create an `UnrecoverableError` we could just cast an existing error i.e. `UnrecoverableError(err)`.

Reply:
Ah I see. I always forget that `error` is an interface and not a struct. In that case, defining `UnrecoverableError` as a concrete type that implements the `error` interface so we can use a type assertion to check if an `error` is an `UnrecoverableError` seems reasonable to me.